### DISCUSSION TITLE
Is it possible to specify GPU order?

### DISCUSSION THREAD

**Discussion by segmond** [ggml-org/llama.cpp#6107] [General]
Is it possible to specify GPU order?   if not, would orders find it useful?

Say I have a system with 4 GPUs 2 3090s and 2 1080's and the 3090's are in slot 1 & 4 so showing up as GPU 0 & 3 and the 1080's are in slot 2 & 3 showing up as 1 & 2.  The reason is because slot 2 & 3 are PCIe 1x whereas slot 1 & 4 are PCIe 8x.     I'll like to prioritize loading models on the 3090's.   Sure with option ts I can do 0, 0 to skip the 1080's but if I have a model that will be bigger than the memory of both 3090's.  I'll like to max them out first before putting anything in the 1080's.   I'll also like to leave them in the slot for when I'm doing training which does need faster PCI bandwidth.

If this is not possible, would other's find it useful to be added?

**Comment by slaren** [comment]
You can select and reorder the CUDA devices visible to the application with the `CUDA_VISIBLE_DEVICES` environment variable, eg. `CUDA_VISIBLE_DEVICES=1,4,0,3`.

**Reply by segmond** [reply]
Thanks very much!
